waku/node/peer_manager/peer_manager.nim

```diff
@@ -412,6 +430,19 @@ proc onPeerEvent(pm: PeerManager, peerId: PeerId, event: PeerEvent) {.async.} =
     direction = if event.initiator: Outbound else: Inbound
     connectedness = Connected
 
+    ## Check max allowed in-relay peers
+    let inRelayPeers = pm.connectedPeers(WakuRelayCodec)[0]
+    if inRelayPeers.len > pm.inRelayPeersTarget and
+        pm.peerStore.hasPeer(peerId, WakuRelayCodec):
+      debug "disconnecting relay peer because reached max num in-relay peers",
+        peerId,
+        inRelayPeers = inRelayPeers.len,
+        inRelayPeersTarget = pm.inRelayPeersTarget
+
+      await pm.switch.disconnect(peerId)
+      pm.peerStore.delete(peerId)
```

Comment:
I would personally only handle the disconnections in `connectToRelayPeers()` or `manageRelayPeers()`, that part of their function is to monitor periodically if we are beyond our targets and prune connections.

I understand making sure we don't exceed the targets in different places, but I think that the role of checking if we are already beyond our target and disconnect from existing peers should happen only in one place.

Maybe I'm missing something. Is there a case in which `connectToRelayPeers()` or `manageRelayPeers()` don't handle the peer targets correctly and we need to do the disconnection over here?